src/libraries/System.Private.CoreLib/src/System/AppDomain.cs

```diff
@@ -278,6 +279,8 @@ public void SetThreadPrincipal(IPrincipal principal)
             }
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2006:UnrecognizedReflectionPattern",
```

Comment:
Same question on RequiresUnreferencedCode.

Reply:
Linker intrinsically recognizes all these weird overloads. This was ported from @marek-safar's  original prototype. I don't think we would add them based on [API usage telemetry](https://apisof.net/catalog/System.AppDomain.CreateInstanceAndUnwrap(String,String,Object())) since nobody uses these APIs.

Yes, my preference would be to make these `string,string` overloads `RequiresUnreferencedCode`, delete the linker handling, and be done with it.

Having to mark these `UnconditionalSuppressMessage` is unfortunate because it puts us into a situation where forgetting to intrinsically recognize this API (e.g. once we build a Roslyn analyzer) will result in potentially broken apps.

With the other annotations in this pull request, if we forget to intrinsically recognize a rare overload, it will result in _more_ warnings, not less warnings.

Reply:
You could start with more aggressive mode and block less APIs based on previews feedback.